_posts/2021-09-23-event-streaming-with-pulsar-and-scala.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "Event Streaming with Apache Pulsar and Scala"
+date: 2021-09-23
+header:
+    image: "/images/blog cover.jpg"
+tags: []
+excerpt: "Apache Pulsar is a cloud-native, distributed messaging and streaming platform that manages hundreds of billions of events per day. We'll learn what makes it a strong competitor to other solutions and use Scala along with the pulsar4s to interact with it."
+---
+
+
+In today's modern data era there is an ever increasing demand for delivering data insights as fast as possible. What happens “now”, might be irrelevant a few minutes or even seconds later and so there is an ever increasing need for events to be ingested and processed as fast as possible, whether it is to improve a business and make it more competitive in a demanding market or make a system grow and adapt itself based on the environment stimulations it receives.
+
+Along with that and the evolution of containers and cloud infrastructure companies seek ways to leverage and adopt cloud-native approaches. 
+Moving to the cloud and adopting containers for our system means we are most likely leveraging technologies like Kubernetes for all of it’s amazing features. 
+Having our infrastructure on the cloud and adopting cloud-native solutions means we also want our messaging and streaming solutions to be in-line with these principles.
+
+In this blog post we will go through how you can implement Cloud Native event stream processing with Apache Pulsar and Scala. We will review what Apache Pulsar has to offer in this modern data era, what makes it stand out and how you can get started with it by creating some simple Producers and Consumers using Scala and the pulsar4s library.
+
+## 1. What Is Apache Pulsar
```

Comment:
The benefits of Pulsar are nice. Because they're now structured as bigger bullets and some notions need some explanation, it's probably worth expanding them a bit and make them subsections of this chapter.